onboard/catkin_ws/src/acoustics/scripts/acoustics_wrapper.py

```diff
@@ -76,19 +94,31 @@ def get_processing_angle(self, wrapper_goal, processing_files, guess):
         return self.client_processing.get_result()
 
     def execute(self, wrapper_goal):
+
         time_now = datetime.now()
         guess_export_name = "date_" + time_now.strftime("%m_%d_%Y_%H_%M_%S") + "_guess"
         processing_export_name = "date_" + time_now.strftime("%m_%d_%Y_%H_%M_%S") + "_processing"
 
-        self.publish_feedback(0, "Starting sampling for guess")
-        guess_sampling = self.saleae_sampling(1, guess_export_name)
-        self.publish_feedback(1, "Sampling for guess complete, starting guess")
+        if wrapper_goal.gen_data == True:
```

Comment:
When you remove this from the wrapper msg, this will become a param. What you might want to do is set a variable that is equal to the param in the init function, so that you don't have to continue trying to get the param, you can just read the instance variable.

Reply:
Edit: You'll be able to remove this if statement completely if you use pass the callback fn as a lambda into this execute fn. See next comment